src/main/app-menu.ts

```diff
@@ -713,8 +722,22 @@ export function setupMenu (
   if (appState?.dimMenu) {
     disableAllMenuItems(template);
   }
-  const menu = Menu.buildFromTemplate(template);
-  Menu.setApplicationMenu(menu);
+
+  // Preserve the submenus as a dedicated array.
+  const submenus = template.map(i => i.submenu);
+  function templateTransform(wnd: BrowserWindow) {
+    return wnd.isFocused() ?
+      (i, idx) => i.submenu = submenus[idx]
+      : (i) => i.submenu = null;
```

Comment:
enabled attribute is ignored for top level menu items, so we take out the submenus from them. This way user won't see the messed up menu popup while windows are transitioning the focus.